packages/angular_devkit/build_angular/src/utils/routes-extractor/extractor.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { ApplicationRef, Injector, Type, createPlatformFactory, platformCore } from '@angular/core';
+import {
+  INITIAL_CONFIG,
+  ɵINTERNAL_SERVER_PLATFORM_PROVIDERS as INTERNAL_SERVER_PLATFORM_PROVIDERS,
+} from '@angular/platform-server';
+import { Route, Router, ɵROUTER_PROVIDERS } from '@angular/router';
+import { first } from 'rxjs/operators'; // Import from `/operators` to support rxjs 6 which is still supported by the Framework.
+
+// TODO(alanagius): replace the below once `RouterConfigLoader` is privately exported from `@angular/router`.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const RouterConfigLoader = ɵROUTER_PROVIDERS[5] as any;
+type RouterConfigLoader = typeof RouterConfigLoader;
+
+interface RouterResult {
+  route: string;
+  success: boolean;
+  redirect: boolean;
+}
+
+async function* getRoutesFromRouterConfig(
+  routes: Route[],
+  routerConfigLoader: RouterConfigLoader,
+  injector: Injector,
+  parent = '',
+): AsyncIterableIterator<RouterResult> {
+  for (const route of routes) {
+    const { path, redirectTo, loadChildren, children } = route;
+    if (path === undefined) {
+      continue;
+    }
+
+    const currentRoutePath = buildRoutePath(parent, path);
+
+    if (redirectTo !== undefined) {
+      // TODO: handle `redirectTo`.
+      yield { route: currentRoutePath, success: false, redirect: true };
+      continue;
+    }
+
+    if (/[:*]/.test(path)) {
+      // TODO: handle parameterized routes population.
+      yield { route: currentRoutePath, success: false, redirect: false };
+      continue;
+    }
+
+    yield { route: currentRoutePath, success: true, redirect: false };
+
+    if (children?.length || loadChildren) {
+      yield* getRoutesFromRouterConfig(
```

Comment:
BTW, as you recurse through child routes, you might need to update the injector. For now, I don't think this will matter so much but technically if a route has an injector, you should use that as the injector when loading child routes.